pkg/config/config.go

```diff
@@ -9,11 +9,12 @@ var (
 	schemas             = field.StringSliceField("schemas", field.WithDefaultValue([]string{"public"}), field.WithDescription("The schemas to include in the sync"))
 	includeColumns      = field.BoolField("include-columns", field.WithDescription("Include column privileges when syncing. This can result in large amounts of data"))
 	includeLargeObjects = field.BoolField("include-large-objects", field.WithDescription("Include large objects when syncing. This can result in large amounts of data"))
+	syncAllDatabases    = field.BoolField("sync-all-databases", field.WithDescription("Sync all databases. This can result in large amounts of data"), field.WithDefaultValue(false))
```

Comment:
nit: I think making this an explicit option is fine, but it would be nice if we had some basic validation on the DSN.

When sync-all-databases is false, the DSN should have the database name provided.
When it is true, the DSN should not have a database name provided.